src/fty_nut_command_server.cc

```diff
@@ -0,0 +1,326 @@
+        return;
+    }
+
+    // Enter mainloop
+    ZpollerGuard poller(zpoller_new(pipe, mlm_client_msgpipe(client), nullptr));
+    zsock_signal(pipe, 0);
+    while (!zsys_interrupted) {
+        void *which = zpoller_wait(poller, -1);
+        if (zsys_interrupted) {
+            break;
+        }
+        if (which == pipe) {
+            ZmsgGuard msg(zmsg_recv(pipe));
+            ZstrGuard actor_command(zmsg_popstr(msg));
+
+            //  $TERM actor command implementation is required by zactor_t interface
+            if (streq(actor_command, "$TERM")) {
+                return;
+            }
+            else if (streq(actor_command, "NUT_SERVER")) {
+                ZstrGuard host(zmsg_popstr(msg));
+                ZstrGuard username(zmsg_popstr(msg));
+                ZstrGuard password(zmsg_popstr(msg));
+                nutHost = host.get();
+                nutUsername = username.get();
+                nutPassword = password.get();
+                log_info("NUT server '%s' configured", host.get());
+                continue;
+            }
+            else if (streq(actor_command, "DB_URL")) {
+                ZstrGuard databaseURL(zmsg_popstr(msg));
+                dbURL = databaseURL.get();
+                log_info("Database URL configured");
+                continue;
+            }
+            else {
+                log_error("Unrecognized pipe request '%s'", actor_command.get());
+                continue;
+            }
+        }
+
+        ZmsgGuard msg(mlm_client_recv(client));
+
+        if (!streq(mlm_client_subject(client), COMMAND_SUBJECT)) {
+            log_error("Unrecognized subject '%s', ignoring message", mlm_client_subject(client));
+            continue;
+        }
+        if (zmsg_size(msg) < 2) {
+            log_error("Message doesn't have UUID and command fields, ignoring message", mlm_client_subject(client));
+            continue;
+        }
+
+        const char *sender = mlm_client_sender(client);
+        ZstrGuard action(zmsg_popstr(msg));
+        ZstrGuard uuid(zmsg_popstr(msg));
+
+        try {
+            // Connect to NUT server
+            nut::TcpClient nutClient;
+            nutClient.connect(nutHost);
+            log_info("Connected to NUT server");
+            nutClient.authenticate(nutUsername, nutPassword);
+            log_info("Authenticated to NUT server");
+
+            // Connect to database
+            tntdb::Connection conn = tntdb::connectCached(dbURL);
+
+            log_info("Received request '%s' from '%s', UUID='%s'", action.get(), sender, uuid.get());
+
+            if (streq(action.get(), "GET_COMMANDS")) {
+                get_commands(nutClient, conn, client, sender, msg, uuid.get());
+            }
+            else if (streq(action.get(), "DO_COMMANDS")) {
+                do_commands(nutClient, conn, client, sender, msg, uuid.get());
+            }
+            else if (streq(action.get(), "ERROR")) {
```

Comment:
Why do we need to handle "ERROR" action command ?

Reply:
We do not want to reply to an `ERROR` message because we risk a ping-pong effect if the other side doesn't ignore `ERROR` messages too.

Reply:
got it